site/docs/v1/tech/apis/api-explorer.adoc

```diff
@@ -0,0 +1,78 @@
+---
+layout: doc
+title: API Explorer
+description: Explore the API for FusionAuth
+---
+
+++++
+<link rel="stylesheet" type="text/css" href="./swagger-ui.css" />
+<style>
+
+/* overrides docs css that causes code text to be dark */
+code.language-json  p {
+color: #FFFFFF;
+}
+code.language-json  span {
+color: #FFFFFF;
+}
+code.language-bash  p {
+color: #FFFFFF;
+}
+code.language-bash  span {
+color: #FFFFFF;
+}
+
+pre.microlight span.headerline {
+color: #FFFFFF;
+}
+
+/* hides smartbear image */
+.topbar-wrapper a.link {
+display:none;
+}
+</style>
+++++
+
+== Overview
+
+This tool uses the https://github.com/fusionauth/fusionauth-openapi[FusionAuth OpenAPI specification] to let you explore APIs FusionAuth offers.
+
+* <<Configuration>>
+** <<In-Browser API Calls>>
+* <<Explorer>>
+
+== Configuration
+
+This tool defaults to using the latest version of the FusionAuth API. If you are using a different version, find the https://github.com/FusionAuth/fusionauth-openapi/tags[correct version] and update the URL in the input box, then click [field]#Explore#.
+
+For example, to view the 1.40.0 release, put `https://raw.githubusercontent.com/FusionAuth/fusionauth-openapi/1.40.0/openapi.yaml` in the YAML file input box.
+
+Don't forget to click on the [field]#Authorize# button to add your FusionAuth API key.
+
+=== In-Browser API Calls
+
+If you want to use the in-browser API explorer, you must enable CORS on your FusionAuth instance. This is not required if you are using the curl commands to make calls against the API from the command line.
+
+To do so, log in to your FusionAuth administrative user interface. Then: 
+
+* Navigate to [breadcrumb]#Settings ->System#
+* Enable [field]#CORS#.
+* Update [field]#Allowed headers# to include `authorization`, `accept` and `content-type` headers.
+* Check all the [field]#Allowed methods#
+* Set [field]#Allowed origins# to your origin. You can also use the value of `*`, which allows any origin to connect.
+
+[WARNING.warning]
+====
+You won't want to run your production FusionAuth instance with CORS as open as this, but for development this is fine.
+====
+
+image::apis/cors-settings-api-explorer.png[CORS settings for the API explorer,width=1200]
+
+== Explorer
+
+++++
+<div id="swagger-ui"></div>
```

Comment:
Are all of the swagger includes stock versions, or did you have to customize them?

Reply:
only had to customize them to use our OpenAPI file. I also had to do some CSS tweaks (the inline <style> block). I could also have created a [custom FusionAuth layout](https://swagger.io/docs/open-source-tools/swagger-ui/customization/custom-layout/) but that seemed like a lot more work.